src/mon/FSCommands.cc

```diff
@@ -1525,6 +1513,32 @@ int FileSystemCommandHandler::_check_pool(
   }
 
   const string& pool_name = osd_map.get_pool_name(pool_id);
+  auto app_map = pool->application_metadata;
+
+  if(!allow_overlay && !app_map.empty()) {
+    auto app = app_map.find(pg_pool_t::APPLICATION_NAME_CEPHFS);
+    if(app != app_map.end()) {
+      auto& [app_name, app_metadata] = *app;
+      auto itr = app_metadata.find("data");
```

Comment:
can the "data" and "metadata" checks be in a single `if` block?

Reply:
Updated (Tried to avoid redundant code).